### ISSUE TITLE
Breaking change in 2.14.0 for Quartz extension with jdbc

### ISSUE THREAD

**Issue by computerlove** [quarkusio/quarkus#29177]
### Describe the bug

When using 
```properties
quarkus.quartz.clustered=true
quarkus.quartz.store-type=jdbc-tx
```
Quartz store job class name in `JOB_DETAILS.JOB_CLASS_NAME`. 
In 2.14.0 `io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob` has been moved to ` io.quarkus.quartz.runtime.QuartzSchedulerImpl$InvokerJob`, so startup crash with
```
io.quarkus.runtime.Application Failed to start application (with profile prod): java.lang.ClassNotFoundException: io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:115)
	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:65)
	at org.quartz.simpl.InitThreadContextClassLoadHelper.loadClass(InitThreadContextClassLoadHelper.java:72)
	at org.quartz.simpl.InitThreadContextClassLoadHelper.loadClass(InitThreadContextClassLoadHelper.java:78)
	at org.quartz.impl.jdbcjobstore.StdJDBCDelegate.selectJobForTrigger(StdJDBCDelegate.java:1662)
	at org.quartz.impl.jdbcjobstore.StdJDBCDelegate.selectJobForTrigger(StdJDBCDelegate.java:1628)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport.replaceTrigger(JobStoreSupport.java:1489)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport$11.execute(JobStoreSupport.java:1478)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport.executeInNonManagedTXLock(JobStoreSupport.java:3864)
	at org.quartz.impl.jdbcjobstore.JobStoreTX.executeInLock(JobStoreTX.java:93)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport.replaceTrigger(JobStoreSupport.java:1474)
	at org.quartz.core.QuartzScheduler.rescheduleJob(QuartzScheduler.java:1121)
	at org.quartz.impl.StdScheduler.rescheduleJob(StdScheduler.java:321)
	at io.quarkus.quartz.runtime.QuartzSchedulerImpl.<init>(QuartzSchedulerImpl.java:294)
	at io.quarkus.quartz.runtime.QuartzSchedulerImpl_Bean.create(Unknown Source)
	at io.quarkus.quartz.runtime.QuartzSchedulerImpl_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
	at io.quarkus.quartz.runtime.QuartzSchedulerImpl_Observer_start_0f88b9afbe74c75c0e6959d7e0bd795b6582f6d3.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:326)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:308)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:76)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:131)
        ...
```

I haven't seen any way this could be mitigated in code, so I think the only way to handle this is to add it to the 2.14.0 migration document.

Running 
```sql
UPDATE JOB_DETAILS set JOB_CLASS_NAME = 'io.quarkus.quartz.runtime.QuartzSchedulerImpl$InvokerJob' WHERE JOB_CLASS_NAME = 'io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob';
```
solves the problem. 

### Expected behavior

Application starts as normal.

### Actual behavior

Application crash with `java.lang.ClassNotFoundException: io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob`

### How to Reproduce?

* Create application using Quarkus 2.13.4 clustered Quartz with jdbc.
* Create a job
* Start application
* Upgrade to 2.14.0 and restart

### Quarkus version or git rev

2.14.0

**Comment by gsmet** [comment]
@machi1990 @mkouba I think we probably need to have a proper API outside of the `runtime` package if it's actually stored in the db. But it's probably too late for that and it doesn't seem wise to break it once again on purpose.

**Comment by gsmet** [comment]
I added a note to the migration guide as suggested here: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.14#quartz .

**Comment by mkouba** [comment]
> @machi1990 @mkouba I think we probably need to have a proper API outside of the `runtime` package if it's actually stored in the db. But it's probably too late for that and it doesn't seem wise to break it once again on purpose.

To be honest, I didn't even know that the name of the job class is stored in the DB. And I have no idea why it's actually stored there. In any case, the `io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob` is definitely not part of the public API.

> I added a note to the migration guide as suggested here: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.14#quartz .

Thanks.

**Comment by gsmet** [comment]
> io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob is definitely not part of the public API.

Well, it's somehow part of an API if it's stored in the DB. Maybe we should add a comment to the enclosing class and the nested class explaining we should avoid renaming them? (And document it in the migration guide if we do)

**Comment by mkouba** [comment]
> > io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob is definitely not part of the public API.
> 
> Well, it's somehow part of an API if it's stored in the DB. Maybe we should add a comment to the enclosing class and the nested class explaining we should avoid renaming them? (And document it in the migration guide if we do)

Ok, let me rephrase the sentence - it's not intentionally part of the API. However, I do understand your concerns.

> Maybe we should add a comment to the enclosing class and the nested class explaining we should avoid renaming them?

Good idea. I had a quick look and it seems like we can't prevent quartz from loading the job class :-(

**Comment by gsmet** [comment]
We added a note in the migration guide. I will close this one.

@mkouba could you add a small comment in the top level class and the nested class so that we avoid renaming them in the future (or document it in the migration guide if there are no other options)? Thanks!

**Comment by mkouba** [comment]
> @mkouba could you add a small comment in the top level class and the nested class so that we avoid renaming them in the future (or document it in the migration guide if there are no other options)? Thanks!

Done: https://github.com/quarkusio/quarkus/pull/29292
